packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -35,6 +40,72 @@ const Indicator = () => {
   const trackedInitialLoad = useRef(false)
   const { track } = useTrackEvent()
 
+  async function fetchPageData() {
+    const urlHostString = window.location.origin
+    const pathAdjustment =
+      window.location.pathname === `/` ? `/index` : window.location.pathname
+
+    const url = `${urlHostString}/page-data${pathAdjustment}/page-data.json`
+    try {
+      const resp = await fetch(url)
+      const data = await resp.text()
+
+      if (data === `{}`) {
+        const err = new Error(`Not Found`)
+        err.status = 404
+        throw err
+      }
+
+      return { data, errorMessage: null }
+    } catch (e) {
+      if (e.status === 404) {
+        return { data: null, errorMessage: `This page has moved.` }
+      } else {
+        return { data: null, errorMessage: null }
+      }
+    }
+  }
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const { data } = await fetchPageData()
+      pageData = data
+    }
+
+    fetchData()
+  }, [])
+
+  const hasPageDataChanged = async () => {
+    if (buildId !== latestCheckedBuild || !pageData) {
+      let pageDataCounter = 0
+      const hasPageChanged = false
+
+      while (!hasPageChanged && pageDataCounter <= PAGE_DATA_RETRY_LIMIT) {
```

Comment:
So once there is a new successful build we try to grab the pagedata up to 30 times, in 1 second intervals until we get it? What if the site takes longer to deploy the routes? For larger sites this can take up to 2 minutes.

Reply:
Do you recommend we poll maybe 2 seconds until 2 minutes? is that 2 minutes after the build is marked as SUCCESS?